src/utils/use-multiselect.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * Hook to deal with list of checkboxes
+ * @param elementIds list of all ids used for selection, whether they are selected or not
+ * for "handleShiftAndCtrlClick" to work, this list needs to be sorted in the same order as it is displayed
+ */
+export const useMultiselect = (elementIds: string[]) => {
+    const [selectedIds, setSelectedIds] = useState<string[]>([]);
+    // used for shift clicking selection, stores last clicked element for selection
+    const [lastSelectedElementId, setLastSelectedElementId] = useState<
+        string | null
+    >(null);
+
+    const clearSelection = () => {
+        setSelectedIds([]);
+    };
+
+    useEffect(() => {
+        clearSelection();
+        setLastSelectedElementId(null);
+    }, [elementIds]);
+
+    /**
+     * toggle selection for one element
+     * @param elementId id of element to toggle
+     * @param forceState if defined, it will force element state instead of toggling (false -> unselect element ; true -> select element)
+     */
+    const toggleSelection = useCallback(
+        (elementToToggleId: string) => {
+            let element = elementIds?.find((id) => id === elementToToggleId);
+            if (element === undefined) {
+                return;
+            }
+
+            selectedIds.indexOf(elementToToggleId);
+            const elementToToggleIdIndex =
+                selectedIds.indexOf(elementToToggleId);
+            // if element to toggle is not selected, we select it
+            if (elementToToggleIdIndex < 0) {
+                selectedIds.push(elementToToggleId);
+            } else {
+                selectedIds.splice(elementToToggleIdIndex, 1);
+            }
+
+            setSelectedIds([...selectedIds]);
+            setLastSelectedElementId(elementToToggleId);
+        },
+        [selectedIds, elementIds]
+    );
+
+    const addElementsToSelection = useCallback(
+        (elementsToSelectIds: string[]) => {
+            elementsToSelectIds
+                .filter(
+                    (elementToSelectId) =>
+                        !selectedIds.includes(elementToSelectId)
+                )
+                .forEach((elementToSelectId) => {
+                    selectedIds.push(elementToSelectId);
+                });
+            setSelectedIds([...selectedIds]);
+        },
+        [selectedIds]
+    );
+
+    const removeElementsFromSelection = useCallback(
+        (elementsToUnselectIds: string[]) =>
+            setSelectedIds([
+                ...selectedIds.filter(
+                    (id) => !elementsToUnselectIds.includes(id)
+                ),
+            ]),
```

Comment:
```suggestion
            const elementsToUnselectIdsSet = new Set(elementsToUnselectIds);
            setSelectedIds([
                ...selectedIds.filter(
                    (id) => !elementsToUnselectIdsSet.has(id)
                ),
            ]),
```
for performance

Reply:
We cannot use Set here because we need to keep order to use lastSelectedElementId ...
My bad